docs/src/AlgebraicGeometry/Schemes/AffineSchemes.md

```diff
@@ -6,229 +6,111 @@ CurrentModule = Oscar
+standard_spec(X::AbsSpec{<:Any, <:MPolyQuoLocalizedRing})
 ```
 
-### A minimal implementation of the `AbsSpecMor` interface
-Again, there is a minimal implementation of this interface for all instances 
-of `AbsSpec` that meet the above requirements on the involved 
-rings (especially the required methods for `hom`):
-```@docs 
-    SpecMor{DomainType<:AbsSpec, 
-            CodomainType<:AbsSpec, 
-            PullbackType<:Hecke.Map
-           }
-```
-The default constructor is
-```
-    SpecMor(X::AbsSpec, Y::AbsSpec, a::Vector{<:RingElem}; check::Bool=true)
-```
-where one has to specify the domain and codomain of the map, and 
-the images `a` of the coordinates (the generators of `ambient_ring(Y)`) 
-under the pullback map ``\mathscr{O}(Y) \to \mathscr{O}(X)``. Expensive checks can be 
-turned off by setting `check=false`.
+### Closed subschemes
 
-The basic functionality for `SpecMor` comprises
+```@docs
+subscheme(X::AbsSpec, f::Vector{<:RingElem})
+subscheme(X::AbsSpec, I::Ideal)
 ```
-   # the composition 
-   compose(f::AbsSpecMor, g::AbsSpecMor) 
 
-   # the identity map
-   identity_map(X::AbsSpec)
+### Intersections
 
-   # restriction to subsets in the domain and codomain
-   restrict(f::AbsSpecMor, X::AbsSpec, Y::AbsSpec; check::Bool=true)
+Support for intersections of affine schemes is
+provided with the command `intersect`.
 
-   # equality test
-   ==(f::AbsSpecMor, g::AbsSpecMor)
+### Open subschemes
 
-   # preimages of subschemes in the codomain
-   preimage(f::AbsSpecMor, Z::AbsSpec)
+```@docs
+hypersurface_complement(X::AbsSpec, f::RingElem)
+hypersurface_complement(X::AbsSpec, f::Vector{<:RingElem})
 ```
-In particular, for every concrete instance of a type `MySpec<:AbsSpec` that 
-implements `underlying_scheme`, this basic functionality of `SpecMor` 
-should run naturally.
 
-### Some particular higher types of morphisms
+### Closure
 
-We may derive higher types of morphisms of affine schemes `MySpecMor<:AbsSpecMor` 
-by storing an instance `g` of `SpecMor` inside an instance `f` of 
-`MySpecMor` and implementing 
-```
-    underlying_morphism(f::MySpecMor)::SpecMor # return g
+```@docs
+closure(X::AbsSpec, Y::AbsSpec)
 ```
-For example, this allows us to define closed embeddings:
+
+
+## Attributes
+
 ```@docs
-    ClosedEmbedding{DomainType, CodomainType, PullbackType}
+OO(X::AbsSpec)
+ambient_ring(X::AbsSpec)
+base_ring(X::AbsSpec)
+dim(X::AbsSpec)
+codim(X::AbsSpec)
+strict_modulus(X::AbsSpec)
 ```
-In addition to the standard getters and methods for instances 
-of `SpecMor`, we also have 
+There are also functions, which return the types of
+schemes and associated rings, as well as
+`defining_ideal`.
+
+
+## Properties
+
 ```@docs
-    image_ideal(f::ClosedEmbedding)
+isempty(X::AbsSpec)
+issubset(X::AbsSpec, Y::AbsSpec)
+is_open_embedding(X::AbsSpec, Y::AbsSpec)
+is_closed_embedding(X::AbsSpec, Y::AbsSpec)
 ```
 
 
+## Methods
+
+With `==`, one can test if two affine schemes are isomorphic.
```

Comment:
Two subschemes `X` and `Y` of the same ambient affine space are considered equal if and only if the identity map 
induces an isomorphism of `X` and `Y`.  This is implemented by `X==Y`. 
For `X` and `Y` in different ambient affine spaces, `X==Y` is always `false`.

Reply:
Thank you. I just added this.